oauth/src/lib.rs

```diff
@@ -162,6 +213,210 @@ fn get_socket_address(redirect_uri: &str) -> Option<SocketAddr> {
     None
 }
 
+/// Struct that handle obtaining and refreshing access tokens.
+///
+/// Should not be instantiate by itself, use [`OAuthClientBuilder`] instead.
```

Comment:
This is redundant as you can only instantiated a `struct` when all fields are public. 
```suggestion
```